src/btree/bt_page.c

```diff
@@ -1783,10 +1794,9 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page, bool *instantiate_updp
              * will be comprised of data that is globally visible, and so the reader thread which
              * attempts to read the page into cache can skip the visible all check.
              */
-            if (!(WT_READING_CHECKPOINT(session) && F_ISSET(session, WT_SESSION_PREFETCH_THREAD)) &&
```

Comment:
I don't think WT_SESSION_PREFETCH_THREAD is related here. Thus I removed this check.

Reply:
Yes, the checking for checkpoint makes sense but no need to check the thread. No matter which thread reads in a page, it should be the same.